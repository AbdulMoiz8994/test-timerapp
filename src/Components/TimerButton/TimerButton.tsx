import React from 'react'

interface Props{
   buttonAction:() => void
   buttonValue: string
}

export const TimerButton: React.FC<Props> = ({buttonAction,buttonValue}) => {
    return (
        <div>
            <button onClick={buttonAction}>{buttonValue}</button>
        </div>
    )
}
