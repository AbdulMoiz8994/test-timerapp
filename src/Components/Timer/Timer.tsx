import React, { useState } from 'react';
import { TimerButton } from '../TimerButton/TimerButton'
export const Timer = () => {

    let [second, setSeconds] = useState<number>(0)
    let [minute, setMinute] = useState<number>(0)
    let [hour, setHour] = useState<number>(0)
    let [status, setStatus] = useState<number>(0);
    let [interv, SetInterval] = useState<any>()
    //start
    const run = () => {
        // In status we are sifting/updating the value because we want stop and start timer again 
        if (status !== 1) {
            setStatus(1)
            // we make the set interval state because we later stop and reset the  timer
            SetInterval(
                setInterval(() => {
                    start()
                }, 1000)
            )
        }

    }
    function start() {
       if(second === 60){
         ++minute
         setMinute(minute)  
         second=0
       }
       if(hour === 60){
           hour++
           setHour(hour)
           minute=0
       }
       setSeconds(++second) 
    }
    // stop
function stop(){
    if(status !== 0){
        setStatus(2)
        clearInterval(interv)
    }
}

    //reset


    return (
        <div>
            <span>{hour < 10 ? `0${hour}`: hour}</span>
            <span>{minute < 10 ? `0${minute}` : minute}</span>
            <span>{second < 10 ? `0${second}` : second}</span>
            <TimerButton
                buttonAction={run}
                buttonValue={"Start"}
            />
            <TimerButton 
             buttonAction={stop}
             buttonValue={"Stop"}
            />
        </div>
    )
}
