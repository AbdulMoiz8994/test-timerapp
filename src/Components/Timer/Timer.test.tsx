import React from 'react';
import {shallow,ShallowWrapper} from 'enzyme'
import {Timer} from './Timer'
import {TimerButton} from '../TimerButton/TimerButton'
describe('Timer',() =>{
    let timer: ShallowWrapper
    beforeEach(() => (timer=shallow(<Timer/>)))
    it('Testing Timer Compnent',() =>{
        expect(timer.find('div').length).toEqual(1)
    })

    it('we have span',() =>{
      const container=shallow(<Timer/>)
      expect(container.find('span').length).toBeGreaterThanOrEqual(3)
    })

    it('we timer button components',() =>{
        const container=shallow(<Timer/>)
         expect(container.find(TimerButton).length).toBeGreaterThanOrEqual(3)  
    })
    it('SnapShot',() =>{
        const container=shallow(<Timer/>)
        expect(container).toMatchSnapshot()
    })
})
