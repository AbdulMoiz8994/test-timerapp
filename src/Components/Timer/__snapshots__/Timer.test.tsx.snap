// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer SnapShot 1`] = `
<div>
  <span>
    00
  </span>
  <span>
    00
  </span>
  <span>
    00
  </span>
  <TimerButton
    buttonAction={[Function]}
    buttonValue="Start"
  />
  <TimerButton
    buttonAction={[Function]}
    buttonValue="Stop"
  />
  <TimerButton
    buttonAction={[Function]}
    buttonValue="ReSet"
  />
</div>
`;
